src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -92,10 +111,22 @@ public String getScript() {
         args.addAll(Arrays.asList("sh", "-c", cmd));
         Launcher.ProcStarter ps = launcher.launch().cmds(args).envs(envVars).pwd(ws).quiet(true);
         listener.getLogger().println("[" + ws.getRemote().replaceFirst("^.+/", "") + "] Running shell script"); // -x will give details
-        /* Uncomment for diagnosis (and comment following line) in case wrapper script fails. Otherwise skip since it consumes a thread:
-        ps.stdout(listener);
-        */
-        ps.readStdout().readStderr(); // TODO RemoteLauncher.launch fails to check ps.stdout == NULL_OUTPUT_STREAM, so it creates a useless thread even if you never called stdout(…)
+        boolean novel;
+        synchronized (encounteredPaths) {
```

Comment:
`synchronizedMap` does not let you atomically increment a value. There may be some utility in Guava etc. that does this.